src/main/java/com/google/sps/servlets/ScheduleServlet.java

```diff
@@ -126,31 +125,46 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   }
 
   /**
-   * Creates a calendar event for each task with the same title, description and
-   * start time. The duration is the default one.
+   * Extends the tasks into a wrapper class that contains also the duration.
+   */
+  List<ExtendedTask> createExtendedTasks(List<Task> tasks, String[] tasksDurations) {
+    List<ExtendedTask> extendedTasks = new ArrayList<>();
+
+    for (ListIterator<Task> tasksIterator = tasks.listIterator(); tasksIterator.hasNext(); ) {
```

Comment:
I wonder if there is some iterator that "zips" two arrays. Otherwise, if doing this manually, you should assert that both arrays have the same length.